src/Profiles.ts

```diff
@@ -141,113 +131,231 @@ export class Profiles {
+        if (schema[value].optionDefinition.hasOwnProperty("defaultValue")){
+            options = {
+                prompt: description,
+                value: schema[value].optionDefinition.defaultValue
+            };
+        } else {
+            options = {
+                placeHolder: description,
+                prompt: description
+            };
         }
+        return options;
+    }
 
-        const quickPickOptions: vscode.QuickPickOptions = {
-            placeHolder: localize("createNewConnection.option.prompt.ru.placeholder", "Reject Unauthorized Connections"),
-            ignoreFocusOut: true,
-            canPickMany: false
-        };
-
-        const selectRU = ["True - Reject connections with self-signed certificates",
-            "False - Accept connections with self-signed certificates"];
-
-        const ruOptions = Array.from(selectRU);
-
-        const chosenRU = await vscode.window.showQuickPick(ruOptions, quickPickOptions);
-
-        if (chosenRU === ruOptions[0]) {
-            rejectUnauthorize = true;
-        } else if (chosenRU === ruOptions[1]) {
-            rejectUnauthorize = false;
-        } else {
-            vscode.window.showInformationMessage(localize("createNewConnection.rejectUnauthorize",
-                "Operation Cancelled"));
-            return undefined;
+    public async createNewConnection(profileName: string): Promise<string | undefined> {
+        let profileType: string;
+        let userName: string;
+        let passWord: string;
+        let port: number;
+        let rejectUnauthorize: boolean;
+        let options: vscode.InputBoxOptions;
+        let isTrue: boolean;
+
+        profileType = await this.getProfileType();
+
+        const schema: {} = await this.getSchema(profileType);
+        const schemaArray = Object.keys(schema);
+
+        const schemaValues: any = {};
+        schemaValues.name = profileName;
+
+        // Go through array of schema for input values
+        for (const value of schemaArray) {
+            switch (value) {
+            case "host" :
+                options = {
+                    placeHolder: localize("createNewConnection.option.prompt.host.placeholder", "host.com"),
+                    prompt: localize("createNewConnection.option.prompt.host",
+                    "Enter a z/OS Host in the format 'host.com'."),
+                };
+                let host = await vscode.window.showInputBox(options);
+                if (!host) {
+                    vscode.window.showInformationMessage(localize("createNewConnection.zosmfURL",
+                        "No valid value for z/OS Host. Operation Cancelled"));
+                    return undefined;
+                }
+                try {
+                    if (host.includes("/") || host.includes(":")) {
+                        const result = this.parseUrl(host);
+                        host = result.host;
+                    }
```

Comment:
> "Discussed to have" meaning it will be added soon? ok then.

@jellypuno, I think @JillieBeanSim was meaning say the code already handles trimming of `https://` and port number if the user includes those in the host URL. I highlighted the code I believe is taking care of this, and I am able to successfully create working profiles while entering URLs that both do and do not contain `https://` and/or port number. If you don't mind sharing, what string are you entering to get the ENOTFOUND error for host URL?

Regarding the wordiness of the field description for base path, I agree the text is a bit on the long side, but at the same time, this is something that would be controlled by the profile schema defined in Zowe CLI (for zosmf) or a Zowe Explorer extender (for other profile types), rather than the code in this PR. Grabbing the field descriptions from these defined profile schemas rather than hard-coding them into the Zowe Explorer code allows more flexibility and scalability/maintainability in the types of profiles that can be created and used with Zowe Explorer.

Billie, please correct me if I'm wrong with my interpretations, thanks.

Reply:
> code already handles trimming of https:// and port number if the user includes those in the host URL. 

@lauren-li I tried doing it today and yes it is working. Couple of things that i saw.

_ If I entered `https://url:port`, I am still going to be ask for a port. If I didn't put anything it will automatically change it to 443. Is it possible to extract the hostname and port if I give the whole URL? 

- If I entered `url:port` (without https) the url ended up being spaces in the yaml file.

Reply:
Thanks @jellypuno for pointing out these cases. I am seeing them as well, and they should be addressed in this PR.

Reply:
> * If I entered `https://url:port`, I am still going to be ask for a port. If I didn't put anything it will automatically change it to 443. Is it possible to extract the hostname and port if I give the whole URL?
> * If I entered `url:port` (without https) the url ended up being spaces in the yaml file.

@jellypuno thanks I will correct this issue,  I tested with `https://...` and that trim was working but not `url:port`.  Not sure what you are asking about the port being included.  Are you wanting that prompt to not be presented if the port was entered in with the host or did you want the port entered within the host to be the value in the textbox vs it showing the default value?

Reply:
hmmm. good point. I was thinking that if I entered `https://url:port`, the port dialog box will not be displayed anymore. But you are right. Maybe it is also good to display for confirmation but instead of 443, it will be the value that I provided. I dont know the better UX for this. I prefer the first one(not to display it). but I don't know. what do you think @lauren-li ?

Reply:
@jellypuno I can see either way working. On one hand, it's more consistent to present the prompt for the port number, using the user's value if they entered it in with the host, or using the default if not. On the other hand, that would make the user have to do an extra step to hit "Enter" again, so it would be more streamlined to just skip the `port number` prompt and use the user's value if they already entered it in as part of the `host` value. I think either option would work.